dist2src/worker/updater.py

```diff
@@ -53,63 +51,69 @@ def check_updates(
         else:
             logger.debug("Celery tasks created never expire")
 
-        m = re.match(
-            r"(?P<fork>forks\/)?((?P<owner>.+)\/)?(?P<namespace>.+)",
-            self.cfg.src_git_namespace,
+        src_gitlab_group = self.cfg.src_git_svc.gitlab_instance.groups.get(
+            self.cfg.src_git_namespace
         )
 
-        # Get repositories in source-git
-        url = f"{self.cfg.src_git_svc.api_url}projects"
-        params = {
-            "namespace": m["namespace"],
-            "pattern": project,
-            "fork": m["fork"] is not None,
-            "per_page": self.PROJECTS_PER_PAGE,
-            "owner": m["owner"] or None,
-            "short": True,
-        }
-        while url:
-            r = self.cfg.src_git_svc.call_api(url, params=params)
-            # The URL in 'next' already has the required parameters
-            url, params = r["pagination"]["next"], None
-            logger.debug(
-                f"Next page: {url!r}. Total pages: {r['pagination']['pages']!r}."
+        # get initial set of projects: if project is defined, about only that one
+        # otherwise update projects in the namespace
+        n = 1
+        if project:
+            projects = [project]
+        else:
+            projects = [
+                p.name
+                for p in src_gitlab_group.projects.list(
+                    page=n, per_page=self.PROJECTS_PER_PAGE
+                )
+            ]
+
+        while projects:
+            for project_to_be_processed in projects:
+                self.check_and_update_project(project_to_be_processed, branch=branch)
+            if project:
+                # there is only a single project to check
+                break
+            n += 1
+            projects = src_gitlab_group.projects.list(
+                page=n, per_page=self.PROJECTS_PER_PAGE
             )
-            for src_git_project in r["projects"]:
-                try:
-                    self._check_project(src_git_project["name"], branch)
-                except (OgrException, RetryError):
-                    logger.exception(
-                        f"Failed checking project {src_git_project['name']!r}"
-                    )
-                    continue
+
         Pushgateway().push_dist2src_finished_checking_updates()
 
-    def _check_project(self, project: str, branch: Optional[str] = None):
-        logger.debug(f"Checking project {project!r}...")
-        dist_git_project = self.cfg.dist_git_svc.get_project(
-            namespace=singular_fork(self.cfg.dist_git_namespace),
-            repo=project,
-            # Specify username in order to disable superfluous whoami API calls.
-            username="packit",
-        )
+    def check_and_update_project(self, project: str, branch: Optional[str] = None):
```

Comment:
Is there a reason to unmark this as "internal use only"? I mean the dropping of the leading underscore.

Reply:
not really, I'll switch it back to private